### PR TITLE
removed circular dep, fixed npm devDep, Angular2-AOT compatibility

### PR DESCRIPTION
**Pull request by metzc**
TODO:

- [-] If this PR is a new feature, reference an issue where a consensus about the design was reached (not necessary for small changes)
- [x] Make sure all of the significant new logic is covered by tests
- [x] Rebase your changes on master so that they can be merged easily
- [x] Make sure all tests and linter rules pass
- [x] Update CHANGELOG.md with your change
- [x] Add your name and email to the AUTHORS file (optional)
- [-] If this was a change that affects the external API, update the docs and post a link to the PR in the discussion

- removed circular dependency in data/store.ts [Issue #731]
- added "ApolloClient" to the named exports to make it compatible with Angular2 AOT compile [Issue #758]
- moved dev @types to devDependencies otherwise they potentially brake projects that are importing apollo-client [Issue #713]

### CODE REVIEW

File: AUTHORS [hunk 1]
```diff
@@ -43,3 +43,4 @@ matt debergalis <matt@meteor.com>
 Vladimir Guguiev <wizardzloy@gmail.com>
 Edvin Eriksson <edvinerikson@gmail.com>
 Agustin Polo <poloagustin@gmail.com>
+Chris Metz <metzc@users.noreply.github.com>
```
Comment (stubailo) [inline AUTHORS]: Aw, not a real email?

File: src/data/storeUtils.ts [hunk 2]
```diff
@@ -34,7 +36,7 @@ function isVariable(value: Value): value is Variable {
   return value.kind === 'Variable';
 }
 
-function isObject(value: Value): value is ObjectValue {
+function isGraphObject(value: Value): value is ObjectValue {
```
Comment (stubailo) [inline src/data/storeUtils.ts]: Maybe `isGraphQLObjectValue` would be a good name. This looks for GraphQL object literals in the AST.

**Comment by stubailo** [comment]
Merging here: https://github.com/apollostack/apollo-client/pull/780

File: AUTHORS (hunk 1 above)
Comment (metzc) [inline AUTHORS]: I wondered.. isn't this openly readable for any email scanner? Also wouldn't put my address anywhere else openly. What do you think?

Comment (stubailo) [inline AUTHORS]: Up to you. I put my real email on GitHub, but I understand not everyone wants to.

**Pull request closed without merging**